earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -74,7 +112,7 @@ DOMDocument *
 CreateEmptyDocument(const std::string &rootTagname) throw()
 {
   InitializeXMLLibrary();
-
+  if (doTerminate) pthread_mutex_lock(&xmlDocLock);
```

Comment:
This line does not check whether the mutex was acquired successfully, or there was an error.